app/views/api/v2/subnets/main.json.rabl

```diff
@@ -6,14 +6,8 @@ attributes :network, :network_type, :cidr, :mask, :priority, :vlanid, :gateway,
            :dns_primary, :dns_secondary,
            :from, :to, :created_at, :updated_at, :ipam, :boot_mode
 
-child :dhcp => :dhcp do
-  extends "api/v2/smart_proxies/base"
-end
-
-child :tftp => :tftp do
-  extends "api/v2/smart_proxies/base"
-end
-
-child :dns => :dns do
-  extends "api/v2/smart_proxies/base"
+Subnet.smart_proxies.keys.each do |proxy|
```

Comment:
`class_attribute` may help here, as its value is inherited by subclasses if a more specific subclass value isn't set. Ses https://github.com/domcleal/foreman/commit/4d05ce2493ed5db55e6b167d43ab816906481e3d for example.

Note that's specifically fetching, merging, then setting the value of smart_proxies_from_model rather than modifying the same hash, which should allow for subclasses to vary their registered smart proxies from the parent (they'll receive a copy of the parent + their own alterations).